rules/prefer-node-append.js

```diff
@@ -1,41 +1,18 @@
 'use strict';
 const getDocumentationUrl = require('./utils/get-documentation-url');
+const isReturnValueUsed = require('./utils/is-return-value-used');
 
 const getMethodName = memberExpression => memberExpression.property.name;
 
-const ignoredParentTypes = [
-	'ArrayExpression',
-	'IfStatement',
-	'MemberExpression',
-	'Property',
-	'ReturnStatement',
-	'VariableDeclarator'
-];
-
-const ignoredGrandparentTypes = [
-	'ExpressionStatement'
-];
-
 const create = context => {
 	return {
 		CallExpression(node) {
-			const {
-				callee,
-				parent
-			} = node;
-
-			const {
-				parent: grandparent
-			} = (parent || {});
+			const {callee} = node;
 
 			if (callee.type === 'MemberExpression' && getMethodName(callee) === 'appendChild') {
 				let fix = fixer => fixer.replaceText(callee.property, 'append');
 
-				if (parent && ignoredParentTypes.includes(parent.type)) {
-					fix = undefined;
-				}
-
-				if (grandparent && ignoredGrandparentTypes.includes(grandparent.type)) {
+				if (!isReturnValueUsed(node)) {
```

Comment:
Seems not right? If used, should has no fix, right?

Reply:
You're right--I inverted it by mistake. Plan to fix now.

Reply:
Fixed.